File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        extracted = 0
        place = 1
        digit_sum = 0
        
        while n > 0:
            rem = n % 10
            digit_sum += rem
            
            if rem != 0:
                extracted = rem * place + extracted
                place *= 10
                
            n //= 10  # Integer division in Python
            
        return extracted * digit_sum

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        long long extracted = 0, place = 1, sum = 0;
        
        while (n > 0) {
            int rem = n % 10;
            sum += rem;
            
            if (rem != 0) {
                extracted = rem * place + extracted; // Builds the number in original order
                place *= 10;
            }
            n /= 10;
        }
        
        return extracted * sum;
    }
};
